ArtOfIllusion/src/artofillusion/MaterialMappingDialog.java

```diff
@@ -39,28 +39,30 @@ public MaterialMappingDialog(BFrame parent, Object3D obj)
   {
     super(parent, "Material Mapping", true);
 
-    fr = parent;
     this.obj = obj;
     map = obj.getMaterialMapping();
     oldMapping = map.duplicate();
 
     // Make a list of all material mappings which can be used for this object and material.
 
-    mappings = new Vector();
-    List<MaterialMapping> allMappings = PluginRegistry.getPlugins(MaterialMapping.class);
-    for (int i = 0; i < allMappings.size(); i++)
+    mappings = new Vector<Class>();
+    Material mat = obj.getMaterial();
+    
+    for(MaterialMapping mapping: PluginRegistry.getPlugins(MaterialMapping.class))
```

Comment:
Thanks for pointing that out. I was going to mention similar. Its fine to use the interface if you are in a situation where several `list` implementations _might_ be supplied, depending on the calling code. In this case, though, its a private member, and you definitely know the concrete type.